blog/2023-10-14-lightweight-cloud-registry-oras/index.mdx

```diff
@@ -0,0 +1,406 @@
+
+Mounting one as a file system **should** be seen as "unintended functionality" by the COS and breaks some assumptions programs make about file systems.  
+
+> Example: Instant nanosecond retrieval times, turning into _millisecond_ or even **second** long affairs.
+
+## Putting it together
+
+Now that we have the all the parts, it's just a matter of combining them!
+
+> For this tutorial we are using s3 with `s3fs`, but feel free to use a different provider and follow along! 
+
+### Prerequisites
+
+- Install [ORAS](https://github.com/oras-project/oras/releases)
+- Install [s3fs](https://github.com/s3fs-fuse/s3fs-fuse#installation)
+- Obtain an AWS account and create a S3 bucket
+
+### Creating some test artifacts
+
+Lets's first create some artifacts!  
+```sh
+echo "Hello World!" > hello.txt
+echo "Goodbye World!" > goodbye.txt
+```
+
+### Mounting your s3 bucket
+
+Now we should mount our s3 bucket so that we can use it as an oci-layout target.  
+Checkout `s3fs --help` for the different ways you can authenticate, here we are using credential profiles.
+```sh
+mkdir s3
+s3fs my-bucket s3 -o profile=MY_AWS_PROFILE
+```
+
+### Using your new "registry"
+
+Pushing our newly created artifacts is as easy as:  
+```sh
+oras push --oci-layout s3/hello:latest hello.txt:text/plain
+oras push --oci-layout s3/goodbye:latest goodbye.txt:text/plain
+```
+
+Pulling them is just as simple!
+```sh
+rm hello.txt
+rm goodbye.txt
+oras pull --oci-layout s3/hello:latest
+oras pull --oci-layout s3/goodbye:latest
+cat hello.txt
+cat goodbye.txt
+```
+
+### Congratulations!
+
+**Now you are running your own lightweight OCI registry on the cloud!**
+
+## Caveats 
+
+<!--bigger warning?-->
+### **You can't delete**
+
+**You can't delete your artifacts!**
+
+Currently[^1] you can't delete your local `oci-layout` artifacts with `oras manifest delete` or `oras blob delete`
+
+The core reason for this is that normally it is the registry's responsibility to garbage collect blobs without any references.
+
+It is still possible to just do `rm -rf s3/my-repo`, but that is probably not what you want to do.
+
+[^1]: It is being worked on in [this issue](https://github.com/oras-project/oras-go/issues/454)
+
+### Concurrency?
+
+Same concurrency rules of your COS apply, generally [last-write-wins](https://en.wikipedia.org/wiki/Eventual_consistency).
+
+
+## Using this trick in your application
+
+You can use this trick in your applications as well!  
+
+Instead of integrating in your application code (hard dependency, vendor lock-in, complicates your interface: "how do I pass in creds?") you instead mount the COS in the container and then "just use the file system" (transferable, simple, external, interfaceable) 
+
+There are some caveats to this approach and your application might leverage the capabilities of the Native COS api extremely well, but if it's just a simple interaction, why complicate things?
+
+
+## Upgrading to Registry
+
+### AWS Simple Storage Service -> ZOT

```

Comment:
What!
We are all AWS ~Marketting~ Foundations Certified!
:^]

I also wanted to be consistent with my naming on all the COS offerings to not use their Acronyms where I could.
It reads more clearly if you are not used to the specific platform.  
I am all ready using the acronym in the steps section so I thought it would be clear.
> if you are coming form pure AWS, do you know what GCS or ABS or OSS refers to? Now imagine it the other way around.